### PR TITLE
Docs: metrics and alerts

### PR DESCRIPTION
**Pull request by jenmalloy**
### Type of change
- Documentation

### Description

Document metrics and alerts

### Checklist
- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [X] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/ref-rules-records.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+`enmasse_address_spaces_not_ready_total{exported_namespace="prod_namespace",service="address-space-controller"} 1`
+
+`enmasse_component_health`::
+Description::: Provides a Boolean-style metric for each `address-space-controller` and `api-server` indicating if they are up and running.
+Expression:::
+[source,options="nowrap"]
+----
+up{job="address-space-controller"} or on(namespace) (1 - absent(up{job="address-space-controller"}))
+up{job="api-server"} or on(namespace) (1 - absent(up{job="api-server"}))
+----
```
Comment (robshelly) [inline documentation/modules/ref-rules-records.adoc]: The other examples and expressions in this module are formatted differently. Maybe they should be formatted ins this style too.

**Comment by robshelly** [comment]
Since this issue was opened we have removed the API server so the ref-metrics-api-server.adoc can be removed.

**Comment by jenmalloy** [comment]
@robshelly, thanks for your review! Addressed your comments (and also added this content to the existing monitoring assembly). Can you please review again? Thanks!

File: documentation/assemblies/assembly-metrics-rules.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+[id='assembly-metrics-rules-{context}]
+= Metrics and rules
+
+include::../modules/ref-metrics-common.adoc[leveloffset=+1]
+
+include::../modules/ref-metrics-address-space-controller.adoc[leveloffset=+1]
+
+include::../modules/ref-metrics-standard-controller-agent.adoc[leveloffset=+1]
+
+include::assemblies/assembly-rules.adoc[leveloffset=+1]
```
Comment (robshelly) [inline documentation/assemblies/assembly-metrics-rules.adoc]: include::./assembly-rules.adoc[leveloffset=+1]

File: documentation/assemblies/assembly-monitoring.adoc [hunk 3]
```diff
@@ -19,5 +19,7 @@ include::../modules/proc-deploy-monitoring-using-ansible.adoc[leveloffset=+1]
 
 include::../modules/proc-configure-alerts.adoc[leveloffset=+1]
 
+include::assemblies/assembly-metrics-rules.adoc[leveloffset=+1]
```
Comment (robshelly) [inline documentation/assemblies/assembly-monitoring.adoc]: include::./assembly-metrics-rules.adoc[leveloffset=+1]

**Comment by jenmalloy** [comment]
@robshelly, Sorry about that, I think I've got it fixed now: http://file.rdu.redhat.com/~jmalloy/openshift/#assembly-metrics-rules-messaging

**Pull request merged**